foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -0,0 +1,277 @@
+                migrateConfiguration = true;
+            }
+            if(containsOrphedomosPackaging(file)){
+                migratePackaging = true;
+            }
+            shouldExecute = migrateConfiguration || migratePackaging;
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true
+     * @param file file to migrate
+     * @return isSuccessful - whether file was migrated successfully or not
+     */
+    @Override
+    protected boolean performMigration(File file) {
+        Model model = getLocationAnnotatedModel(file);
+        Plugin plugin  = getPlugin(model, OLD_ARTIFACT_ID);
+        boolean replacedConfig = false;
+        boolean replacedPackaging = false;
+        boolean isSuccessful = false;
+
+        Xpp3Dom config;
+
+        if (plugin != null){
+            config = getConfiguration(plugin);
+        } else {
+            config = null;
+            logger.info("Orphedomos plugin configuration not found. Skipping configuration migration.");
+        }
+
+        // check if pom file has orphedomos-maven-plugin configuration
+        if (config != null){
+            // grab the old imageVersion and imageName
+            IMAGE_VERSION = getImageVersion(config);
+            IMAGE_NAME = getImageName(config);
+
+            // Replace groupID and artifactID
+            PomModifications modifyPlugin = new PomModifications();
+            InputLocation startGroupId = plugin.getLocation("groupId");
+            InputLocation endGroupId = PomHelper.incrementColumn(startGroupId, OLD_GROUP_ID.length());
+
+            InputLocation startArtifactId = plugin.getLocation("artifactId");
+            InputLocation endArtifactId = PomHelper.incrementColumn(startArtifactId, OLD_ARTIFACT_ID.length());
+
+            modifyPlugin.add(new PomModifications.Replacement(startGroupId, endGroupId, NEW_GROUP_ID));
+            modifyPlugin.add(new PomModifications.Replacement(startArtifactId, endArtifactId, NEW_ARTIFACT_ID));
+
+            writeModifications(file, modifyPlugin.finalizeMods());
+
+            Model updatedModel = getLocationAnnotatedModel(file);
+            Plugin updatedPlugin = getPlugin(updatedModel, NEW_ARTIFACT_ID);
+
+            if (IMAGE_VERSION != null && IMAGE_NAME != null) {
+                PomModifications modifyImages = new PomModifications();
+                // add new configuration code
+                int indent = 4;
+                int startConfigCodeRow = updatedPlugin.getLocation("artifactId").getLineNumber();
+                InputLocation startConfigCode = new InputLocation(startConfigCodeRow + 1, updatedPlugin.getLocation("artifactId").getColumnNumber());
+                modifyImages.add(new PomModifications.Insertion(startConfigCode, indent, OrphedomosToFabric8Migration::getConfigCode));
+
+                replacedConfig = writeModifications(file, modifyImages.finalizeMods());
+
+            }
```

Comment:
Q: it seems like we are wiping out their existing orphedomos config, and just replacing with the generic fabric8 config. Did we have a rationale for that being fine?

Reply:
@Cho-William that's a good point.. I decided to go with this method because of the fact that we can't do a 1:1 migration. I figured if users had multiple different configurations and we kept it, it could potentially break the pom entirely.

Reply:
If I recall, Emily said this is fine, so I don't think we have any action to do here tbh

Reply:
sounds good, keeping as is